src/nix-daemon/nix-daemon.cc

```diff
@@ -278,14 +279,26 @@ static int _main(int argc, char * * argv)
                 printVersion("nix-daemon");
             else if (*arg == "--stdio")
                 stdio = true;
-            else return false;
+            else if (*arg == "--trust") {
```

Comment:
What does this flag do? What are we trusting?

Reply:
`stdin`. When we have a Unix domain socket we can do `getPeerInfo`, but we can't do that with stdin so currently we just always trust `stdin` (that is, if we aren't just proxying stdin another daemon, as is the common case with `ssh-ng`).

This provides a way to *not* trust stdin, which is useful for testing. I could make these flag `--trust-stdin` /  `--no-trust-stdin` for clarity.

Reply:
I have generalized the flag so it also works in the Unix domain socket case; it forces the trust one way or the other instead of checking the user via`authPeer`.